include/seqan3/search/fm_index/bi_fm_index_cursor.hpp

```diff
@@ -908,9 +908,24 @@ class bi_fm_index_cursor
                       "The alphabet types of the given text and index differ.");
         assert(index != nullptr);
 
-        size_type const loc = offset() - index->fwd_fm.index[fwd_lb];
-        size_type const query_begin = loc - index->fwd_fm.text_begin_rs.rank(loc + 1) + 1; // Substract delimiters
-        return text | views::join | views::slice(query_begin, query_begin + query_length());
+        // Position of query in concatenated text.
+        size_type const location = offset() - index->fwd_fm.index[fwd_lb];
+
+        // The rank represents the number of delimiters before position `location + 1`.
+        // Furthermore, this number represents the i-th text this (text) position originates from, since each delimiter
+        // corresponds to one text in the whole concatenated text from left-to-right.
+        size_type const rank = index->fwd_fm.text_begin_rs.rank(location + 1);
+        assert(rank > 0);
+        size_type const text_id = rank - 1;
+
+        // The sum of all text lengths prior to the i-th text is the start location of the i-th text in the concatenated
+        // sequence.
```

Comment:
Always state the important part, the result, first and then the cause, for a better understanding :)
And technically we are not computing the sum of lengths but `select` just asks "where is the position of the `rank`-th 1".
```suggestion
        // The start location of the `text_id`-th text in the sequence (position of the `rank`-th 1 in the bitvector).
```

Reply:
I think this is worse. Sometime you have to paraphrase what the intend or effect is and not what it is technically doing.

Reply:
I think it is always easier to understand if the comment starts with "what is this" and then "why is it computed like this".
And for me it is actually confusing why the intend should be to compute the "sum of all text lengths"? when I actually just want to know "Where is this 1 that I computed in the rank query beforehand" which is my start position of the text.
Do you know rank and select queries? This is a very typical use case.